src/utils/apollo.ts

```diff
@@ -1,8 +1,28 @@
-import {gql} from '@apollo/client';
-import {client} from '../../App';
+import {ApolloClient, createHttpLink, gql, InMemoryCache} from '@apollo/client';
 import {GraphQLServerError} from '../utils/custom-error';
 import {Alert} from 'react-native';
-import {saveOnAsyncStorage} from '../utils/storage';
+import {getStoredItem, saveOnAsyncStorage} from '../utils/storage';
+import {setContext} from '@apollo/client/link/context';
+
+const BASE_URL =
+  'https://template-onboarding-node-sjz6wnaoia-uc.a.run.app/graphql';
+const link = createHttpLink({uri: BASE_URL});
+
+const authenticationContext = setContext(async (_, {headers}) => {
+  let token = await getStoredItem('token');
+  token = token.replace(/['"]+/g, '');
+  return {
+    headers: {
+      ...headers,
+      Authorization: token ? token : '',
+    },
+  };
```

Comment:
já que token pode não estar disponível n vale a pena colocar essa atualização direto depois de validar se existe? pode ocorrer um erro ao tentar fazer o replace se token n existe

```suggestion
  return {
    headers: {
      ...headers,
      Authorization: token ? token.replace(/['"]+/g, '') : '',
    },
  };
```